src/libcore/panicking.rs

```diff
@@ -36,7 +36,34 @@
                       and related macros",
             issue = "0")]
 
+#[cfg(not(stage0))]
+use any::Any;
 use fmt;
+#[cfg(not(stage0))]
+use panic::{Location, PanicInfo};
+
+// NOTE This function never crosses the FFI boundary; it's a Rust-to-Rust call
+#[cfg(not(stage0))]
+#[allow(improper_ctypes)] // PanicInfo contains a trait object which is not FFI safe
+extern "Rust" {
+    #[lang = "panic_impl"]
+    fn panic_impl(pi: &PanicInfo) -> !;
+}
+
+#[cfg(not(stage0))]
+#[cold] #[inline(never)]
+pub fn panic_payload<M>(msg: M, file_line_col: &(&'static str, u32, u32)) -> !
```

Comment:
I forget, but is this something we want to enable for libcore? I thought that we didn't have a path to actually supporting this yet?

(support in the sense of transmitting the value all the way to the caught value of `thread::spawn`)

Reply:
> I forget, but is this something we want to enable for libcore?

Well, this (adding payload support to core::panic!) *is* in the accepted RFC. Though we don't have to add it right now; we can stabilize #[panic_implementation] w/o payload support and add payload support later on in a backwards compatible fashion (or at least I think it's possible).